lib/modules/manager/argocd/extract.ts

```diff
@@ -66,28 +115,13 @@ function createDependency(
   };
 }
 
-export function extractPackageFile(
-  content: string,
-  fileName: string,
-  _config?: ExtractConfig
-): PackageFileContent | null {
-  // check for argo reference. API version for the kind attribute is used
-  if (fileTestRegex.test(content) === false) {
-    return null;
-  }
-
-  let definitions: ApplicationDefinition[];
-  try {
-    definitions = loadAll(content) as ApplicationDefinition[];
-  } catch (err) {
-    logger.debug({ err, fileName }, 'Failed to parse ArgoCD definition.');
-    return null;
+export function processMultipleSources(
+  sources: Array<ApplicationSource>
+): Array<PackageDependency | null> {
+  const deps: Array<PackageDependency | null> = [];
+  for (const source of sources) {
+    deps.push(processSingleSource(source));
   }
 
-  const deps = definitions
-    .filter(is.plainObject)
-    .map((definition) => createDependency(definition))
-    .filter(is.truthy);
-
-  return deps.length ? { deps } : null;
+  return deps;
```

Comment:
Here the returned type is an `object` and not an `array`, no?
So wouldn't it be better to just do: 
```typescript
  return { deps };
```

Reply:
Because the force pushing Github has become confused. This belongs to 
https://github.com/renovatebot/renovate/pull/20648/files#diff-de5a018e5a8529dc329235872d68a919609bbdbb0e15bf7d19821fea28df153aR121
as part of the return value of `processAppSpec` to prevent that null values are returned.